google/cloud/bigquery/external_config.py

```diff
@@ -1003,3 +1003,104 @@ def from_api_repr(cls, resource: dict) -> "ExternalConfig":
         config = cls(resource["sourceFormat"])
         config._properties = copy.deepcopy(resource)
         return config
+
+
+class ExternalCatalogDatasetOptions(object):
+    """Options defining open source compatible datasets living in the BigQuery catalog.
+    Contains metadata of open source database, schema or namespace represented
+    by the current dataset.
+
+    Args:
+        default_storage_location_uri: Optional. The storage location URI for all
```

Comment:
NIT:

add type annotations for init arguments too.

Reply:
I am not sure what we mean by this.
The init arguments on lines 1023 and 1024 have type hinting as do init args on lines 1072-1074.
(I don't know if that is the result of my continuing to muck around with changes while this comment was being generated OR if you are requesting something else that I am just not understanding).

Please advise. Thanks.

Reply:
Gotcha. will see what we can do.